### PR TITLE
T254975: Improve Library Card embed

### PR DESCRIPTION
**Pull request by bharatswamy**
[//]: # (Thank you for uploading a PR to the Wikipedia Library!)

## Description
Added open graph meta tags to beautify the Wikipedia library embed link 

## Rationale
[//]: # (Why is this change required? What problem does it solve?)
When a link to the Library Card platform homepage is embedded on another website, such as Facebook or Slack, the result isn't pretty,

## Phabricator Ticket
[//]: # (Link to the Phabricator ticket)
https://phabricator.wikimedia.org/T254975

## How Has This Been Tested?
[//]: # (- Did you add tests to your changes? Did you modify tests to accommodate your changes?)
[//]: # (- Can this change be tested manually? How?)
Tested by deploying to the local server and pasting the link to open graph debugger by Facebook
https://developers.facebook.com/tools/debug/

## Screenshots of your changes (if appropriate):
[//]: # (It can also be a GIF to prove that your changes are working)
image

## Types of changes
What types of changes does your code introduce? Add an `x` in all the boxes that apply:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Minor change (fix a typo, add a translation tag, add section to README, etc.)

### CODE REVIEW

File: TWLight/templates/new_base.html
Comment (jsnshrmn) [inline TWLight/templates/new_base.html:19]: In `TWLight/templates/login_partial.html`, you can see how we template a source message with the correct arguments here:
```
{% comment %}Translators: Information about The Wikipedia Library.{% endcomment %}
{% blocktranslate trimmed %}
Over 90 of the world's top subscription-only databases,
with content in {{ no_of_languages }} languages, 
free for Wikipedians of all backgrounds
{% endblocktranslate %}
```
For proper i18n, I think you need to:
- use the above message block for `og:description`
- loop through the available languages, and set each one as an `og:locale:alternate` 
- add support for setting the locale via a url parameter (which might be a big task to do correctly; I'd need to do a little research)

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
Thank you for your work in this! As we add new features, it's really important that we consider how those features will be localized, and I think there is some work to do here. This approach currently is English only, and also hardcodes the language count, which will quickly go out of date. See my inline comment for more details.

If you're wanting to dig deeper into django, i18n handling can open up lots of interesting lines of inquiry, IMO. Please let me know if you have questions or comments!

File: TWLight/templates/new_base.html
Comment (bharatswamy) [reply TWLight/templates/new_base.html:19]: > loop through the available languages, and set each one as an og:locale:alternate

Hi @jsnshrmn to use this we will need list of `language_TERRITORY` refer: [https://ogp.me/#locale](https://ogp.me/#array:~:text=og%3Alocale%20%2D%20The%20locale%20these%20tags%20are%20marked%20up%20in.%20Of%20the%20format%20language_TERRITORY.%20Default%20is%20en_US.)

we can get a list of languages from the `Language` model in the resources app but it does not have `TERRITORY` do we populate it somewhere in our code or should I try fetching it using some third-party package eg: https://github.com/flyingcircusio/pycountry or is there another way

Comment (jsnshrmn) [reply TWLight/templates/new_base.html:19]: That's a great question. Generally, only the language is required in XPG formatted locale names, eg. `language[_territory[.codeset]][@modifier]` where everything in brackets is optional. I would start with our locales as they are and see if that works.

Because of our integration with translatewiki, we're already crosswalking Django's locales with wikimedia CLDR data, so we could add some machinery to define default territories if necessary, but I wouldn't bother with that complexity unless it's clear that it's required for interoperability.

We do have a few locales with territories defined already that you can use for testing in case they are required:
- en_Gb
- ks-arab
- pt_Br
- sr_Ec
- zh_Hans
- zh_Hant

Comment (bharatswamy) [reply TWLight/templates/new_base.html]: Thank you for the precise explanation @jsnshrmn I have made changes according to your suggestion and pushed a new commit

File: TWLight/templates/new_base.html
Comment (jsnshrmn) [inline TWLight/templates/new_base.html:13]: I missed this before, but this should be localized as well.
```suggestion
      {% comment %}Translators: Name of the project/tool.{% endcomment %}
    <meta property="og:title" content="{% trans "The Wikipedia Library" %}" />
```

File: TWLight/views.py
Comment (jsnshrmn) [inline TWLight/views.py:127]: Exclude current language here (and rename to something like alternate_languages)

File: TWLight/templates/new_base.html
Comment (jsnshrmn) [inline TWLight/templates/new_base.html:29]: use the value from [django.utils.translation.get_language_from_request()](https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.translation.get_language_from_request) instead of hardcoding `en_US` Django will take care of figuring out the best locale to use.

**Review by jsnshrmn — CHANGES_REQUESTED** [review]
Followup on my previous remarks. Thanks again for your work, this is almost there!

File: TWLight/templates/new_base.html
Comment (bharatswamy) [reply TWLight/templates/new_base.html]: Oh, I missed this as well, thank you for pointing it out. I have added the trans tag in the latest commit

File: TWLight/templates/new_base.html
Comment (bharatswamy) [reply TWLight/templates/new_base.html]: Thanks, I have removed the hardcoded `en_US` and used the current language here

File: TWLight/templates/new_base.html
Comment (bharatswamy) [reply TWLight/templates/new_base.html]: done, I have excluded the current language from this list

File: TWLight/templates/new_base.html
Comment (bharatswamy) [reply TWLight/templates/new_base.html]: Thank you for the review I have made the changes according to the suggestions using the [django.utils.translation.get_language_from_request()](https://docs.djangoproject.com/en/3.2/ref/utils/#django.utils.translation.get_language_from_request). This has really simplified the task

**Review by jsnshrmn — COMMENTED** [review]
Sorry for the long wait on this review! The code looks good based on a read-through. I'm going to test this in our staging environment as the last step before approval as long as there are no problems.

**Review by jsnshrmn — APPROVED** [review]

**Review by jsnshrmn — APPROVED** [review]
Okay, I tested this with a few languages, and it looks good. Thanks for all of your work!

**Pull request merged**
